src/application/job_manager.py

```
@@ -0,0 +1,41 @@
+        status: str, 
+        metadata: Optional[Dict[str, Any]] = None
+    ) -> None:
+        async with self._lock:
+            if job_id in self._jobs:
+                job = self._jobs[job_id]
+                job.status = status
+                if metadata:
+                    job.metadata.update(metadata)
+                job.metadata['updated_at'] = datetime.utcnow().isoformat()
+
+    async def list_jobs(self, status: Optional[str] = None) -> List[DownloadJob]:
```

Ruff (0.12.2)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)